### PR TITLE
Better server

### PR DESCRIPTION
**Pull request by lucaschimweg**
Added a better server :)

### CODE REVIEW

File: collect_data.py [hunk 1]
```diff
@@ -2,22 +2,28 @@
...
 def main():
+    vid_handler = VideoHandler()
+    key_handler = KeyHandler()
     with Webcam() as cam, \
-         ControllerServer(port=8000) as controller, \
-         VideoServer(port=8050) as video_server, \
-         ImageDiskWriter(folder="collected_data") as img_disk_writer, \
-         CSVDiskWriter(filename="collected_data/classes.csv") as csv_disk_writer:
+        Server(handlers={
+            "/": FileHandler("index.html"),
```
Comment (jstriebel) [inline collect_data.py]: I would prefer to have `FileHandler(…)` above with the other two handlers to keep it consistent.

File: handlers/video_handler.py [hunk 2]
```diff
@@ -1,39 +1,15 @@
+from handlers.handler import Handler
+from aiohttp import web, WSMsgType
+import asyncio 
 import cv2
-import asyncio
-from aiohttp import web
-from threading import Thread
-
 from writers.writer import Writer
```
Comment (jstriebel) [inline handlers/video_handler.py]: Please order the imports like this:
1. Alphabetically all external libraries
2. Newline
3. Alphabetically all internal imports

File: handlers/video_handler.py [hunk 3]
```diff
@@ -63,3 +39,12 @@ def video_handler(self, request):
...
         return resp
+
+    def __enter__(self):
+        self.running = True
+
+    def __exit__(self, exit_type, value, traceback):
+        self.running = False
+
+    def write(self, data):
+        self.frame = data
```
Comment (jstriebel) [inline handlers/video_handler.py]: Please add the final newline

**Review by jstriebel — CHANGES_REQUESTED** [review]
Generally LGTM, just some minor issues that I commented in the code.

**Comment by lucaschimweg** [comment]
@jstriebel I fixed the problems, could you have a look at the code again?

**Comment by jstriebel** [comment]
This is fine from my side, @normanrz, could you have a look, too?

**Review by jstriebel — APPROVED** [review]

**Review by normanrz — APPROVED** [review]
lgtm

**Pull request merged**